app/src/main/java/io/gnosis/safe/utils/ImageUtils.kt

```diff
@@ -15,25 +15,32 @@ fun ImageView.loadTokenLogo(
     @DrawableRes backgroundDrawable: Int? = R.drawable.circle
 ) {
     setPadding(0)
+    setImageDrawable(null)
+    colorFilter = null
     background = backgroundDrawable?.let {
         ContextCompat.getDrawable(context, backgroundDrawable)
     }
-    setImageDrawable(null)
-    colorFilter = null
     when {
         icon == "local::native_currency" -> {
             setImageResource(R.drawable.ic_native_logo)
         }
-        icon?.startsWith("local::") == true -> setImageResource(placeHolderResource)
-
-        !icon.isNullOrBlank() ->
+        icon?.startsWith("local::") == true -> {
+            setImageResource(placeHolderResource)
+        }
+        !icon.isNullOrBlank() -> {
+            //FIXME: workaround, as Picasso uses a final ApplicationContext for its #getDrawable() action, which will not listen to theme's attributes.
+            // this would lead to day/night mode not handled correctly
+            val placeholderDrawable = ContextCompat.getDrawable(context, placeHolderResource)!!
```

Comment:
Is this really a workaround? Seems like you are loading the resource correctly

Reply:
it would be nice to just pass resource id's directly to picasso, without loading them first